src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -104,19 +104,75 @@ public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int to
         return query(topK, null, null, null, id, namespace, filter, false, false);
     }
 
+    @Override
+    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK,
+                                                                              String id,
+                                                                              String namespace,
+                                                                              boolean includeValues,
+                                                                              boolean includeMetadata) {
+        return query(topK, null, null, null, id, namespace, null, includeValues, includeMetadata);
+    }
+
     @Override
     public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK,
                                                                               String id,
                                                                               String namespace) {
         return query(topK, null, null, null, id, namespace, null, false, false);
     }
 
+    @Override
+    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK, String id,
+                                                                              boolean includeValues,
+                                                                              boolean includeMetadata) {
+        return query(topK, null, null, null, id, null, null, includeValues, includeMetadata);
+    }
+
     @Override
     public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK,
                                                                               String id) {
         return query(topK, null, null, null, id, null, null, false, false);
     }
 
+
+    @Override
+    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVector(int topK, List<Float> vector,
+                                                                            String namespace, Struct filter,
+                                                                            boolean includeValues,
+                                                                            boolean includeMetadata) {
+        return query(topK, vector, null, null, null, namespace, filter, includeValues, includeMetadata);
+    }
+
+    @Override
+    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVector(int topK, List<Float> vector,
+                                                                            String namespace, Struct filter) {
+        return query(topK, vector, null, null, null, namespace, filter, false, false);
```

Comment:
Not saying this is wrong or right,  but OOC why do you allow users to have a `filter` when they query by vector, but not when they query by ID? I'd say they both probs would benefit from the inclusion of a `filter`, since users who use these types of queries (I'd imagine) are looking for similarity clusters, essentially. So getting all vectors similar to a given vector (or a given ID), filtered by `xyz` would be helpful

Reply:
I think we support what you're talking ab out on ln 100 in this file, am I misunderstanding?

```
    @Override
    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK,
                                                                              String id,
                                                                              String namespace,
                                                                              Struct filter) {
        return query(topK, null, null, null, id, namespace, filter, false, false);
    }

```